// @tamagui/react-native-web-internals/dist/cjs/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
}, __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var index_exports = {};
__export(index_exports, {
  AccessibilityUtil: () => import_AccessibilityUtil.AccessibilityUtil,
  ImageLoader: () => import_ImageLoader.ImageLoader,
  InteractionManager: () => import_InteractionManager.InteractionManager,
  Platform: () => import_Platform.Platform,
  StyleSheet: () => import_StyleSheet2.StyleSheet,
  TextAncestorContext: () => import_TextAncestorContext.TextAncestorContext,
  TextInputState: () => import_TextInputState.TextInputState,
  UIManager: () => import_UIManager.UIManager,
  canUseDOM: () => import_canUseDOM.canUseDOM,
  colorProps: () => import_colorProps.colorProps,
  createDOMProps: () => import_createDOMProps.createDOMProps,
  createEventHandle: () => import_createEventHandle.createEventHandle,
  createReactDOMStyle: () => import_createReactDOMStyle.createReactDOMStyle,
  createSheet: () => import_dom.createSheet,
  dismissKeyboard: () => import_dismissKeyboard.dismissKeyboard,
  flattenStyle: () => import_StyleSheet.flatten,
  getBoundingClientRect: () => import_getBoundingClientRect.getBoundingClientRect,
  isSelectionValid: () => import_isSelectionValid.isSelectionValid,
  isWebColor: () => import_isWebColor.isWebColor,
  multiplyStyleLengthValue: () => import_multiplyStyleLengthValue.multiplyStyleLengthValue,
  normalizeColor: () => import_normalizeColor.normalizeColor,
  pick: () => import_pick.pick,
  processColor: () => import_processColor.processColor,
  requestIdleCallback: () => import_requestIdleCallback.requestIdleCallback,
  setValueForStyles: () => import_setValueForStyles.setValueForStyles,
  stylesFromProps: () => import_createDOMProps2.stylesFromProps,
  unitlessNumbers: () => import_unitlessNumbers.unitlessNumbers,
  useElementLayout: () => import_useElementLayout.useElementLayout,
  useEvent: () => import_useEvent.useEvent,
  useHover: () => import_useHover.useHover,
  useLayoutEffect: () => import_useLayoutEffect.useLayoutEffectImpl,
  useMergeRefs: () => import_useMergeRefs.useMergeRefs,
  usePlatformMethods: () => import_usePlatformMethods.usePlatformMethods,
  useStable: () => import_useStable.useStable
});
module.exports = __toCommonJS(index_exports);
__reExport(index_exports, require("./modules/AssetRegistry/index"), module.exports);
__reExport(index_exports, require("./modules/forwardedProps/index"), module.exports);
__reExport(index_exports, require("./modules/mergeRefs/index"), module.exports);
__reExport(index_exports, require("./modules/modality/index"), module.exports);
__reExport(index_exports, require("./modules/useLocale/index"), module.exports);
var import_usePlatformMethods = require("./modules/usePlatformMethods/index"), import_TextAncestorContext = require("./TextAncestorContext");
__reExport(index_exports, require("@tamagui/react-native-use-pressable"), module.exports);
__reExport(index_exports, require("@tamagui/react-native-use-responder-events"), module.exports);
var import_colorProps = require("./colorProps"), import_AccessibilityUtil = require("./modules/AccessibilityUtil/index"), import_canUseDOM = require("./modules/canUseDOM"), import_createDOMProps = require("./modules/createDOMProps/index"), import_createDOMProps2 = require("./modules/createDOMProps/index"), import_createReactDOMStyle = require("./StyleSheet/compiler/createReactDOMStyle"), import_createEventHandle = require("./modules/createEventHandle/index"), import_dismissKeyboard = require("./modules/dismissKeyboard/index"), import_getBoundingClientRect = require("./modules/getBoundingClientRect/index"), import_ImageLoader = require("./modules/ImageLoader/index"), import_isSelectionValid = require("./modules/isSelectionValid/index"), import_isWebColor = require("./modules/isWebColor/index"), import_multiplyStyleLengthValue = require("./modules/multiplyStyleLengthValue/index"), import_normalizeColor = require("./modules/normalizeColor/index"), import_pick = require("./modules/pick/index"), import_Platform = require("./modules/Platform/index");
__reExport(index_exports, require("./StyleSheet/preprocess"), module.exports);
var import_StyleSheet = require("./StyleSheet/index"), import_dom = require("./StyleSheet/dom/index"), import_requestIdleCallback = require("./modules/requestIdleCallback/index"), import_setValueForStyles = require("./modules/setValueForStyles/index"), import_TextInputState = require("./modules/TextInputState/index"), import_UIManager = require("./modules/UIManager/index"), import_unitlessNumbers = require("./modules/unitlessNumbers/index"), import_useElementLayout = require("./modules/useElementLayout/index"), import_useEvent = require("./modules/useEvent/index"), import_useHover = require("./modules/useHover/index"), import_useLayoutEffect = require("./modules/useLayoutEffect/index"), import_useStable = require("./modules/useStable/index"), import_InteractionManager = require("./modules/InteractionManager");
__reExport(index_exports, require("./modules/invariant"), module.exports);
var import_processColor = require("./modules/processColor/index"), import_StyleSheet2 = require("./StyleSheet/index"), import_useMergeRefs = require("./modules/useMergeRefs/index");
