// @tamagui/react-native-web-lite/dist/esm/without-animated.mjs
import { createElement } from "./createElement/index.mjs";
import { processColor, Platform, UIManager, canUseDOM, dismissKeyboard, isWebColor, mergeRefs, useMergeRefs, useEvent, useHover, ImageLoader } from "@tamagui/react-native-web-internals";
import { render } from "./render/index.mjs";
import { NativeModules } from "./NativeModules/index.mjs";
import { default as default2 } from "./vendor/react-native/EventEmitter/NativeEventEmitter.mjs";
import { AccessibilityInfo } from "./AccessibilityInfo/index.mjs";
import { Alert } from "./Alert/index.mjs";
import { Appearance } from "./Appearance/index.mjs";
import { AppRegistry } from "./AppRegistry/index.mjs";
import { AppState } from "./AppState/index.mjs";
import { BackHandler } from "./BackHandler/index.mjs";
import { Clipboard } from "./Clipboard/index.mjs";
import { DeviceInfo } from "./DeviceInfo/index.mjs";
import { DeviceEmitter } from "./DeviceEmitter.mjs";
import { DeviceEmitter as DeviceEmitter2 } from "./DeviceEmitter.mjs";
import { Dimensions } from "./Dimensions/index.mjs";
import { I18nManager } from "./I18nManager/index.mjs";
import { Keyboard } from "./Keyboard/index.mjs";
import { Linking } from "./Linking/index.mjs";
import { PanResponder } from "./PanResponder/index.mjs";
import { PixelRatio } from "./PixelRatio/index.mjs";
import { Share } from "./Share/index.mjs";
import { Vibration } from "./Vibration/index.mjs";
import { UnimplementedView } from "./UnimplementedView.mjs";
import { UnimplementedView as UnimplementedView2 } from "./UnimplementedView.mjs";
import { UnimplementedView as UnimplementedView3 } from "./UnimplementedView.mjs";
import { UnimplementedView as UnimplementedView4 } from "./UnimplementedView.mjs";
import { UnimplementedView as UnimplementedView5 } from "./UnimplementedView.mjs";
import { UnimplementedView as UnimplementedView6 } from "./UnimplementedView.mjs";
import { UnimplementedView as UnimplementedView7 } from "./UnimplementedView.mjs";
import { TouchableOpacity } from "./TouchableOpacity.mjs";
import { TouchableWithoutFeedback } from "./TouchableWithoutFeedback.mjs";
import { ActivityIndicator } from "./ActivityIndicator/index.mjs";
import { Image } from "./Image/index.mjs";
import { ImageBackground } from "./ImageBackground/index.mjs";
import { KeyboardAvoidingView } from "./KeyboardAvoidingView/index.mjs";
import { Modal } from "./Modal/index.mjs";
import { Pressable } from "./Pressable/index.mjs";
import { RefreshControl } from "./RefreshControl/index.mjs";
import { SafeAreaView } from "./SafeAreaView/index.mjs";
import { ScrollView } from "./ScrollView/index.mjs";
import { StatusBar } from "./StatusBar/index.mjs";
import { Text } from "./Text/index.mjs";
import { TextInput } from "./TextInput/index.mjs";
import { View } from "./View/index.mjs";
import { LogBox } from "./LogBox/index.mjs";
import { useColorScheme } from "./useColorScheme/index.mjs";
import { useLocaleContext } from "./useLocaleContext/index.mjs";
import { useWindowDimensions } from "./useWindowDimensions/index.mjs";
export * from "@tamagui/react-native-web-internals";
function requireNativeComponent(name) {
  return function () {
    return null;
  };
}
export { AccessibilityInfo, ActivityIndicator, Alert, AppRegistry, AppState, Appearance, BackHandler, Clipboard, DeviceEmitter, DeviceEmitter2 as DeviceEventEmitter, DeviceInfo, Dimensions, UnimplementedView as DrawerLayoutAndroid, UnimplementedView4 as FlatList, I18nManager, Image, ImageBackground, ImageLoader, Keyboard, KeyboardAvoidingView, Linking, LogBox, Modal, default2 as NativeEventEmitter, NativeModules, PanResponder, PixelRatio, Platform, Pressable, RefreshControl, SafeAreaView, ScrollView, UnimplementedView7 as SectionList, Share, StatusBar, UnimplementedView2 as Switch, Text, TextInput, UnimplementedView5 as TouchableHighlight, UnimplementedView6 as TouchableNativeFeedback, TouchableOpacity, TouchableWithoutFeedback, UIManager, Vibration, View, UnimplementedView3 as VirtualizedList, canUseDOM, dismissKeyboard, isWebColor, mergeRefs, processColor, render, requireNativeComponent, createElement as unstable_createElement, useColorScheme, useEvent, useHover, useLocaleContext, useMergeRefs, useWindowDimensions };
